### ISSUE TITLE
GlusterFS support

### ISSUE THREAD

**Issue by rgooch** [Symantec/Dominator#613]
@cviecco I'm thinking about some implementation questions. I'll put each one in a separate comment so that each can be replied to independently.

First, should VM owners specify which Gluster/volume to use for the backing store, or should they just specify that they want any GusterFs volume?

**Comment by rgooch** [comment]
@cviecco Secondly, should a common set of credentials be distributed to all Hypervisors, or should they be partitioned so that different sets of credentials are distributed to different groups of Hypervisors?

**Comment by cviecco** [comment]
I am assuming each hypervisor will have its own set of TLS certs. But I could be convinced otherwise

**Comment by rgooch** [comment]
No, the Hypervisors share the same client certificate. It's not a very powerful certificate. These are the permitted methods:
`Hypervisor.PrepareVmForMigration
ImageServer.CheckDirectory
ImageServer.FindLatestImage
ImageServer.GetImage
ObjectServer.GetObjects`
